src/transformers/models/whisper/tokenization_whisper_fast.py

```diff
@@ -296,6 +296,17 @@ def decode(
         Returns:
             `str`: The decoded sentence.
         """
+        token_ids = to_py_obj(token_ids)
+        prompt_token_id = self.convert_tokens_to_ids("<|startofprev|>")
+        has_prompt = isinstance(token_ids, list) and prompt_token_id in token_ids
+        # If an initial prompt was used, we need to remove it when skipping special tokens
+        if has_prompt and skip_special_tokens:
+            for i in range(1, len(token_ids)):
+                prompt_end_idx = i
+                if token_ids[i] in self.all_special_ids:
+                    break
+            token_ids = token_ids[prompt_end_idx:]
+
```

Comment:
Since this is the same logic as in the regular tokenizer, maybe we can extract it into a shared helper function?

Reply:
moved some of the functionality into a helper and left some for what I think is the right reusability/readability tradeoff, lmk if you think more should be abstracted